ports/coreMQTT/azure_iot_core_mqtt.c

```diff
@@ -0,0 +1,233 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_core_mqtt.c
+ * @brief Implements the port for Azure IoT MQTT based on coreMQTT.
```

Comment:
I know it's in header but may be worth calling out again that the interface at level is subject to change.

When we get really serious about locking down this level's contract we should link back to some sort of MQTT porting guide doc, but that is in the future at this point.